Find a functionally equivalent code:android.bluetooth.BluetoothDevice.fetchUuidsWithSdp:COMMENT
Method Modifier: public      
Comment:/**
 * Perform a service discovery on the remote device to get the UUIDs supported.
 *
 * <p>This API is asynchronous and {@link #ACTION_UUID} intent is sent,
 * with the UUIDs supported by the remote end. If there is an error
 * in getting the SDP records or if the process takes a long time,
 * {@link #ACTION_UUID} intent is sent with the UUIDs that is currently
 * present in the cache. Clients should use the {@link #getUuids} to get UUIDs
 * if service discovery is not to be performed.
 * <p>Requires {@link android.Manifest.permission#BLUETOOTH}.
 *
 * @return False if the sanity check fails, True if the process
 * of initiating an ACL connection to the remote device
 * was started.
 */

@@ -1,6 +1,11 @@
 {
+    IBluetooth service = sService;
+    if (service == null) {
+        Log.e(TAG, "BT not enabled. Cannot fetchUuidsWithSdp");
+        return false;
+    }
     try {
-        return sService.fetchRemoteUuids(this);
+        return service.fetchRemoteUuids(this);
     } catch (RemoteException e) {
         Log.e(TAG, "", e);
     }

